plugins/woocommerce/templates/cart/cart.php

```diff
@@ -85,23 +87,9 @@
 						<td class="product-name" data-title="<?php esc_attr_e( 'Product', 'woocommerce' ); ?>">
 						<?php
 						if ( ! $product_permalink ) {
-							/**
-							 * Filter the product name.
-							 *
-							 * @since 7.8.0
-							 * @param string $product_name Name of the product in the cart.
-							 * @param array $cart_item The product in the cart.
-							 * @param string $cart_item_key Key for the product in the cart.
-							 */
-							echo wp_kses_post( apply_filters( 'woocommerce_cart_item_name', $product_name, $cart_item, $cart_item_key ) . '&nbsp;' );
+							echo wp_kses_post( $product_name . '&nbsp;' );
 						} else {
-							/**
-							 * Filter the product name.
-							 *
-							 * @since 7.8.0
-							 * @param string $product_url URL the product in the cart.
-							 */
-							echo wp_kses_post( apply_filters( 'woocommerce_cart_item_name', sprintf( '<a href="%s">%s</a>', esc_url( $product_permalink ), $product_name ), $cart_item, $cart_item_key ) );
+							printf( '<a href="%s">%s</a>', esc_url( $product_permalink ), wp_kses_post( $product_name ) );
```

Comment:
@rrennick @helgatheviking — this is a *lot* cleaner than how things used to be ... but ... the previous convention (ie, what we had before 7.8.0, where the hook [existed twice](https://github.com/woocommerce/woocommerce/blob/7.0.0/plugins/woocommerce/templates/cart/cart.php#L79-L83) and supplied a product name in one case and a link in another) was in place for a long time.

Can we consider the impact on existing third party code doing something like the following?

```php
add_filter( 'woocommerce_cart_item_name', function () {
	return '<a href="foo-bar">Foo Bar</a>';
} );
```

This is not a hypothetical, as I found code doing essentially this (though of course I've simplified it greatly) when searching the marketplace. This would not have presented a problem prior to 7.8.0, regardless of [which branch](https://github.com/woocommerce/woocommerce/blob/7.0.0/plugins/woocommerce/templates/cart/cart.php#L79-L83) was followed in relation to our hook. Now, however, the above code would run the risk of output looking like:

```html
<a href="PRODUCTURL"><a href="foo-bar">Foo Bar</a></a>
```

What do we think?

Reply:
@barryhughes Yup. We need to revert the filters that pass the `<a>` tags back to the pre 7.8.0 implementation. The test code in the description would add ` - test` after the link pre 7.8.0 while 7.8.0 adds it to the link text and after the link.

Reply:
Yep, sorry now I re-read your earlier comments I see I'm repeating the same thing.